app/models/friendship.rb

```diff
@@ -1,4 +1,23 @@
 class Friendship < ApplicationRecord
+  # Callback
+  # after_create :create_inverse_relationship
+  # after_destroy :destroy_inverse_relationship
+
+  # Associations
   belongs_to :user
   belongs_to :friend, class_name: 'User'
+
+  # Validations
+  validates :user_id, uniqueness: { scope: :friend_id }
```

Comment:
Guys. I recommend going further with this uniqueness enforcement. The situation I propose you cover is the ff:

Suppose `ryto` initiated a friendship with `mike` (`ryto.friendships.create(friend: mike)`). We don't want to allow neither another running of `ryto.friendships.create(friend: mike)` nor `mike.friendships.create(friend: ryto)` to happen successfully. Meaning if `ryto` and `mike` are already friends, we want to prevent either of them from ever creating a duplicate friendship with the other. :) 

We can do the ff:

- Have a private method in here that checks whether there is already an existing friendship between `user_id` and `friend_id`. If yes then simply `return`, otherwise this private method will swap `user` with `friend` and `friend` with `user`:

```ruby
def swap_if_no_record_found
  return unless #there is a friendship record found between user_id and friend_id
  user, friend = friend, user
end
```

- Then having a `before_save` callback here that will invoke the above private method. What this does is force the database to have a conflict if indeed there is an existing friendship between the `user` and `friend` in the friendship record we want to save.